<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>642</width>
    <height>797</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Kênh an toàn sử dụng RSA</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>631</width>
      <height>741</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Tạo key</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>591</width>
        <height>381</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="title">
       <string>Tạo khoá</string>
      </property>
      <widget class="QTextBrowser" name="text_p">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>p = </string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_q">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>70</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>q = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>60</y>
         <width>41</width>
         <height>16</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>&gt;&gt;</string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_n">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>30</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>70</y>
         <width>71</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>0n =</string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_phi_n">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>70</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>30</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>n = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>60</y>
         <width>41</width>
         <height>16</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>&gt;&gt;</string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_e">
       <property name="geometry">
        <rect>
         <x>480</x>
         <y>30</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>70</y>
         <width>71</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>d =</string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_d">
       <property name="geometry">
        <rect>
         <x>480</x>
         <y>70</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>30</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>e = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Public key (n,e) : </string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_public_key">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>130</y>
         <width>401</width>
         <height>61</height>
        </rect>
       </property>
      </widget>
      <widget class="QTextBrowser" name="text_private_key">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>220</y>
         <width>401</width>
         <height>61</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Private key (n,d) : </string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_TaoKhoa">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>310</y>
         <width>181</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Tạo khoá</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Giải mã </string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>601</width>
        <height>711</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="title">
       <string>Giải mã</string>
      </property>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>40</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Nhập n: </string>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_Input_n">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>40</y>
         <width>261</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_Input_d">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>90</y>
         <width>261</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>90</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Nhập d: </string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>149</y>
         <width>581</width>
         <height>211</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="title">
        <string>Nội dung bản mật</string>
       </property>
       <widget class="QTextBrowser" name="text_content_BanMa02">
        <property name="geometry">
         <rect>
          <x>5</x>
          <y>20</y>
          <width>451</width>
          <height>181</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="btn_chose_BanMa">
        <property name="geometry">
         <rect>
          <x>460</x>
          <y>20</y>
          <width>111</width>
          <height>71</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>Chọn file bản mật</string>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>380</y>
         <width>581</width>
         <height>211</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="title">
        <string>Nội dung bản rõ</string>
       </property>
       <widget class="QTextBrowser" name="text_content_BanRo02">
        <property name="geometry">
         <rect>
          <x>15</x>
          <y>20</y>
          <width>551</width>
          <height>181</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QPushButton" name="btn_GiaiMa">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>640</y>
         <width>181</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Giải mã</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Mã hoá</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>601</width>
        <height>701</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="title">
       <string>Mã hoá</string>
      </property>
      <widget class="QPushButton" name="btn_MaHoa">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>640</y>
         <width>181</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Mã hoá</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>390</y>
         <width>591</width>
         <height>221</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="title">
        <string>Nội dung bản mã hoá</string>
       </property>
       <widget class="QTextBrowser" name="text_content_BanMa">
        <property name="geometry">
         <rect>
          <x>15</x>
          <y>20</y>
          <width>561</width>
          <height>181</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>8</pointsize>
         </font>
        </property>
       </widget>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>90</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Nhập e: </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>40</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Nhập n: </string>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_Input_n_2">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>40</y>
         <width>261</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_Input_e">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>90</y>
         <width>261</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>581</width>
         <height>231</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="title">
        <string>Nội dung bản rõ</string>
       </property>
       <widget class="QPushButton" name="btn_chose_BanRo">
        <property name="geometry">
         <rect>
          <x>470</x>
          <y>40</y>
          <width>101</width>
          <height>61</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>Chọn file bản rõ</string>
        </property>
       </widget>
       <widget class="QTextBrowser" name="text_content_BanRo">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>451</width>
          <height>161</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>642</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
